### ISSUE TITLE
/media-source/* are disabled/flaky in mozilla chromium (encoding-specific tests)

### ISSUE THREAD

**Issue by zcorpan** [web-platform-tests/wpt#11181]
http://bocoup.github.io/wpt-disabled-tests-report/

Investigate what's up with this test:

Path | Products | Results | Bugs
-- | -- | -- | --
/media-source/mediasource-config-change-mp4-a-bitrate.html | mozilla chromium |   disabled: [ Failure Pass ] | https://bugzilla.mozilla.org/show_bug.cgi?id=1130973 https://crbug.com/807110

cc @wolenetz @acolwell

The problem here seems to be with tests that test a specific encoding and platforms that don't support playback of those. Correct? What can be done?

**Comment by zcorpan** [comment]
@jyavenard, can you look into this?

**Comment by wolenetz** [comment]
For the upcoming changeType tests (full prototype of Chromium code + new wpt tests: https://chromium-review.googlesource.com/c/chromium/src/+/1093719), I auto-generate tests depending on what subset of test media is supported by the user agent running the test; I also include a test case that fails if a minimum bar isn't met. This approach, IMHO, is better than these mediasoure-config-change-*, because it ensures there is spec-compliance for covered streams, and also that a minimum level of some stream is supported. Inspecting the dynamically generated test cases shows the reader which supported types were tested and their result.

tl;dr: I think a rework of the mediasource-config-change-* suite along similar lines would prove beneficial.
@jyavenard, I can take on this work later, if you want.
